### ISSUE TITLE
Bug: Gutter Width set to insanely large inline amounts

### ISSUE THREAD

**Issue by Aetherinox** [mayurankv/Obsidian-Code-Styler#147]
When adding large amounts of code to a single code box, the gutter width will push off-screen when you scroll down to the bottom of the code box and start clicking the lines.

### Steps to reproduce

- Create new code block, no code block properties needed.
- Add a large amount of code.
 - In my example, I had almost 600 lines
- Scroll to the bottom and click one of the last lines
- To reset the gutter width, click on another page and then come back to the page with the code block

### Expected behaviour

Codebox should keep same gutter size the entire way scrolling down.

### Current behaviour

When scrolling down to the bottom of a rather large codeblock, left side gutter width auto adjusts and pushes text off screen.

### Environment

- Plugin Version: v1.0.11
- Obsidian Version:  1.4.12
- Platform:  Desktop
- OS:  Windows
- Theme:  Default / Blue Topaz
- CSS Snippets:  None

### Screenshots

![FxXUnRS](https://github.com/mayurankv/Obsidian-Code-Styler/assets/118329232/de4ce827-1103-4583-96ba-6a664993ed1b)

### Additional context

When inspecting element, each gutter line has an inline style width set which changes to insanely large numbers.
In the example below, the gutter width is 400px +
![IH4czsN](https://github.com/mayurankv/Obsidian-Code-Styler/assets/118329232/fb683062-13c8-46d9-82fd-4084e14402e0)

**Comment by mayurankv** [comment]
Huh, that is super weird. I'm quite busy with work right now but I'll definitely prioritise looking into this when I can in a couple of weeks. I'll update this thread when I've had a look. Thanks for the bug report!

**Comment by Aetherinox** [comment]
Ah added note. I was messing around with it again earlier today, and it was doing it for a codeblock with as little as 70 lines. Still not sure what is causing it to trigger.

If the codeblock only has a few lines, it's fine.

**Comment by Aetherinox** [comment]
Here's an update since I did more testing:

I tried all of this out on a stock install of Obsidian. Default theme, absolutely zero plugins installed, and the issue still happens.

As soon as I scroll down to the bottom of a code block, the text starts shoving itself to the right. 

I've tried to trigger it every-which-way I can. The issue doesn't happen if you do a regular code block with no language defined. As soon as I add `bash` to the code block, it starts shifting.

I was trying to see if a particular character was making it worse or not, but that doesn't seem to be the case, sometimes it acts like it's all based on how many characters are in the block all-together. But sometimes it triggers with more or less.

At this point, it's completely random, and I can't find the actual root cause.

**Comment by mayurankv** [comment]
Just to double check this is in editing mode right?

**Comment by mayurankv** [comment]
Sorry for the delay, uni is pretty busy right now so I'll probably only have time around Christmas to work on this. Sorry about this. I have a good idea of why this is occuring but I need a more robust solution around the little hack causing this. If you find anything else, please do let me know. Appreciate the detailed report.

**Comment by Aetherinox** [comment]
> Just to double check this is in editing mode right?

Yeah, I'm in Edit mode 99.9% of the time. Tbh, I completely forgot about preview mode. I just went into that mode and checked, and the code blocks are indeed fine and are static. As soon as I flip to edit mode and scroll down, the text starts to run.

**Comment by mayurankv** [comment]
For clarity I'm pretty sure this is related to #17 about how to calculate the gutter width for line numbers in editing mode and pertains to lines 56-63 and 311 in `EditingView.ts`.

If you have any suggestions on an alternative way to calculate this, that would be very welcome but I'll definitely have a think about how I can switch this out!

By any chance could you add a sample file or text for a sample file where the issue occurs?

**Comment by mayurankv** [comment]
Really sorry about this, as I said I'm unfortunately quite busy with work right now but I'll really try and find some time to get at least a temporary fix out there!

**Comment by mayurankv** [comment]
I'm going to close this for now as I think it's done but please let me know if any issues persist!

**Comment by Aetherinox** [comment]
Praise the dev gods. I just opened Obsidian and saw the update. I'll test it out. Thanks!

**Comment by Aetherinox** [comment]
I just pasted a 1000 line piece of code, and thus far, it looks great! Thank friggen god.
I was having to use the Code Files plugin for any snippets of code over like 100 lines and it was making crap get thrown all over the place.

Thanks again. Early Christmas gift!

**Comment by mayurankv** [comment]
Haha sorry for the long wait is all I can say. Very glad I found the time to get it done! Thank you for all the help!
